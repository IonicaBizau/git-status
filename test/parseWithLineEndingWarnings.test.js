const parseWithLineEndingWarnings = require('../lib').parseWithLineEndingWarnings;
const gitStatusOutputSample = ' M example/index.js\0  M lib/index.js\0  M package.json\0 ?? .idea/';
const lineEndingsError = `
warning: LF will be replaced by CRLF in path/to/x/y.z
The file will have its original line endings in your working directory.
`.trim();
test('should return unchanged result if no error', () => {
    parseWithLineEndingWarnings((err, data) => {
        expect(err).toBe(null);
        expect(data).toMatchSnapshot();
    })(null, [{x: '', y: '', to: '', from: ''}]);
});

test('should give back only an error if spawno fails', () => {
    parseWithLineEndingWarnings((err, data) => {
        expect(err.message).toBe('oh');
        expect(data).toBe(undefined);
    })(new Error('oh'), '');
});

test('should parse result if only line error warnings found', () => {
    parseWithLineEndingWarnings((err, data) => {
        expect(err).toBe(null);
        expect(data).toMatchSnapshot();
    })(lineEndingsError, gitStatusOutputSample);
});

test('should provide error if not only line error warnings found', () => {
    parseWithLineEndingWarnings((err, data) => {
        expect(err).toMatchSnapshot();
        expect(data).toBe(undefined);
    })(lineEndingsError + '\nOh snap!', gitStatusOutputSample);
});