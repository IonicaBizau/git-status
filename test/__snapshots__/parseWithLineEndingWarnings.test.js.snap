exports[`test should parse result if only line error warnings found 1`] = `
Array [
  Object {
    "from": null,
    "to": "example/index.js",
    "x": " ",
    "y": "M",
  },
  Object {
    "from": null,
    "to": " lib/index.js",
    "x": " ",
    "y": " ",
  },
  Object {
    "from": null,
    "to": " package.json",
    "x": " ",
    "y": " ",
  },
  Object {
    "from": null,
    "to": " .idea/",
    "x": " ",
    "y": "?",
  },
]
`;

exports[`test should provide error if not only line error warnings found 1`] = `
[Error: warning: LF will be replaced by CRLF in path/to/x/y.z
The file will have its original line endings in your working directory.
Oh snap!]
`;

exports[`test should return unchanged result if no error 1`] = `
Array [
  Object {
    "from": "",
    "to": "",
    "x": "",
    "y": "",
  },
]
`;
